include/meta/corpus/corpus.h

```diff
@@ -75,6 +75,16 @@ class corpus
     virtual document next() = 0;
 
     /**
+     * Reset the corpus to the beginning.
+     */
+    virtual void reset() = 0;
+
+    /**
+     * Skip the next n documents in this corpus (default = 1).
+     */
+    virtual void skip(uint64_t n = 1) = 0;
```

Comment:
Correct. Do you see any value in leaving in there or would you prefer to remove it?